Improve /operational-dasboard

rth opened the pull request:
Improves the operational dashboard (previously was known as  `/plot`).

This endpoint is expected to show the situation at a given time. 

WIP because I still want to add a table with key indicators on the left. 
There is a selector by region, because of https://github.com/icubam/icubam/issues/154 for now the full page is reloaded when region is changed.

If interested I can send you screenshots by DM.

rth commented on icubam/backoffice/handlers/operational_dashboard.py:
```diff
@@ -0,0 +1,151 @@
+        logging.warning(
+          f"Invalid query argument: region={arg_region} "
+          f"Falling back to all regions."
+        )
```
I know I should add tests for this path, but for I haven't managed to make it work (user needed to be monkeypatched, and I was likely missing something).

olivierteboul replied:
you can just do like the other with mock.patch. Then you can try different requests  with different regions..

Check  backoffice/server_test.py

rth commented on icubam/backoffice/templates/operational-dashboard.html:
```diff
@@ -0,0 +1,85 @@
+                 <div class="card">
+                    <div class="card-body">
+		       <h3 class="card-title">{{ metric['value'] }}</h3>
+		       <p class="card-text">{{ metric['label'] }}</p>
```
Would you know a formatter for jinja2 files by any chance? Struggling with tabs here :/

olivierteboul replied:
yeah. it s bit messy but no worries.

olivierteboul commented on icubam/backoffice/handlers/operational_dashboard.py:
```diff
@@ -0,0 +1,151 @@
+  async def get(self):
+    """Serves a page with a table gathering current bedcount data with some extra information."""
+    arg_region = self.get_query_argument('region', default=None)
+    current_user = self.get_current_user()
```
You actually have access to `self.user`

olivierteboul commented on icubam/backoffice/handlers/operational_dashboard.py:
```diff
@@ -0,0 +1,151 @@
+
+    figures = []
+
+    bed_counts = self.db.get_visible_bed_counts_for_user(current_user.user_id)
```
use self.user instead of current_user

olivierteboul commented on icubam/backoffice/handlers/operational_dashboard.py:
```diff
@@ -0,0 +1,151 @@
+    bed_counts = to_pandas(bed_counts)
+
+    if current_region is not None:
+      mask = bed_counts['icu_region_id'] == current_region.region_id
```
i would check that,  `if not self.user.is_admin`  then we should make sure that the requested region_id is part of the region_ids of the managed_icus.

olivierteboul replied:
we do not want the a user to have access to regions he does not belong

olivierteboul commented on icubam/backoffice/handlers/operational_dashboard.py:
```diff
@@ -0,0 +1,153 @@
+  ROUTE = '/operational-dashboard'
+
+  @tornado.web.authenticated
+  async def get(self):
```
it is a very long function so i would split the logic of building the plot knowing that you have the correct dataFrame in another function and call it here. it would be more readable.

rth commented:
OK I have made the following changes,
 - split main method into multiple functions to help readability
 - Added localization. @olivierteboul Do you know what I need to do to see this page in french? I am currently starting the server as `python scripts/run_server.py --config=config.toml --server=backoffice` where   `config.toml` does have `default_locale='fr-fr'` (but my browser has English locale by default). Haven't localized the plot labels (they are all in French) for now.
 - Skip the test for `?region=<id>` queries, that fail for reasons unrelated to this PR (something about how we monkeypatch user)
 - Made aggregation by department instead of city, as requested so it's less sensitive.
 - Made this endpoint only accessible by admin. Until we figure out permissions. Essentially this dashboard shows information aggregated by region / department. If we only display information from ICUs that a given user has access to this aggregated information will be wrong. Also I still think filtering should be done at the level of DB query not handler.

@olivierteboul I would like to merge this as is for now, then iteratively improve it. In any case this is an admin only page for now.
